# Gurgaon AC Dataset

### By Harshul Gupta
### harshul.18bci1055@abes.ac.in

__Importing the required libraries__
- Pandas for exploring the dataset
- Plotly for data visualization (plotly is used because it provides better visualization than mataplotlib and seaborn)

In [1]:
import plotly.offline as py
py.init_notebook_mode(connected=True)
import pandas as pd
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import plotly.express as px
import plotly.subplots as sp

__Reading the dataset__

In [2]:
data = pd.read_csv("AC_Data.csv")

In [3]:
data.shape

(87840, 19)

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87840 entries, 0 to 87839
Data columns (total 19 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       87840 non-null  object 
 1   AC 1    43708 non-null  float64
 2   AC 2    43711 non-null  float64
 3   AC 3    43713 non-null  float64
 4   AC 4    43704 non-null  float64
 5   AC 5    43716 non-null  float64
 6   AC 6    43717 non-null  float64
 7   AC 7    43679 non-null  float64
 8   AC 8    43704 non-null  float64
 9   AC 9    43654 non-null  float64
 10  AC 10   43690 non-null  float64
 11  AC 11   43670 non-null  float64
 12  AC 12   43713 non-null  float64
 13  AC 13   43719 non-null  float64
 14  AC 14   43696 non-null  float64
 15  AC 15   43715 non-null  float64
 16  AC 16   43710 non-null  float64
 17  AC 17   43714 non-null  float64
 18  AC 18   43723 non-null  float64
dtypes: float64(18), object(1)
memory usage: 12.7+ MB


- The dataset contains data of 18 ACs along with the date time of the time 
- It has a total of 19 attributes and 87840 tuples
- Also the dataset has entries for every 2 mins, i.e., the data gets added first at 12:00 p.m. and then gets a new entry at 12:02 p.m.

In [5]:
data.head()

0      AC 1      AC 2  AC 3  AC 4      AC 5      AC 6  \
0  2019-08-01 00:00:00  7.518632  8.788315   0.0   0.0  2.617045  4.079041   
1  2019-08-01 00:01:00       NaN       NaN   NaN   NaN       NaN       NaN   
2  2019-08-01 00:02:00  7.426114  8.940615   0.0   0.0  2.581625  3.781231   
3  2019-08-01 00:03:00       NaN       NaN   NaN   NaN       NaN       NaN   
4  2019-08-01 00:04:00  7.052986  9.161103   0.0   0.0  2.592095  3.800127   

       AC 7      AC 8      AC 9     AC 10     AC 11     AC 12     AC 13  \
0  2.782276  4.624447  5.222060  2.151238  1.585072  0.560373  3.142941   
1       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
2  2.529366  5.057423  5.349465  2.414715  2.168184  1.818730  3.085110   
3       NaN       NaN       NaN       NaN       NaN       NaN       NaN   
4  2.332304  6.322521  3.995392  2.237114  3.345624  2.310409  3.132799   

      AC 14     AC 15     AC 16     AC 17     AC 18  
0  2.749470  5.417774  4.113460  3.305072  6.735981  
1       NaN       NaN       NaN       NaN       NaN  
2  2.720484  3.302422  3.986483  3.220588  6.379500  
3       NaN       NaN       NaN       NaN       NaN  
4  2.676861  3.539026  3.797881  3.131560  6.363475

__For every odd min the data is null value, thus these tuples need to be dropped__

In [6]:
data.describe()

AC 1          AC 2          AC 3          AC 4          AC 5  \
count  43708.000000  43711.000000  43713.000000  43704.000000  43716.000000   
mean       2.379368      3.868892      2.568411      2.395158      1.421699   
std        2.276782      2.545147      1.487508      1.797529      0.767218   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      2.123263      1.706508      1.258299      0.970744   
50%        2.094330      3.791752      2.486992      2.308961      1.380183   
75%        3.661752      5.460892      3.441527      3.476123      1.870234   
max       11.194789     11.147141      8.430909      9.879500      5.495358   

               AC 6          AC 7          AC 8          AC 9         AC 10  \
count  43717.000000  43679.000000  43704.000000  43654.000000  43690.000000   
mean       2.518138      2.980883      2.548711      3.184382      1.538344   
std        1.655560      1.473498      1.504482      1.522291      0.685461   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        1.543779      2.190803      1.766230      2.294635      1.117139   
50%        2.665063      3.005071      2.608551      3.082134      1.504554   
75%        3.616988      3.897857      3.496358      4.096374      1.963707   
max        9.929291      9.356776      9.331506     10.026747      4.625030   

              AC 11         AC 12         AC 13         AC 14         AC 15  \
count  43670.000000  43713.000000  43719.000000  43696.000000  43715.000000   
mean       2.406141      2.699639      4.279893      2.917122      4.046283   
std        1.554481      1.436087      1.847168      1.454946      2.016898   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        1.591748      1.949729      3.260674      1.852981      3.023501   
50%        2.351361      2.727152      4.217381      2.787084      4.001279   
75%        3.254937      3.566744      5.379832      3.824467      5.260193   
max       10.363151      9.887306      9.796393      9.054969     10.349998   

              AC 16         AC 17         AC 18  
count  43710.000000  43714.000000  43723.000000  
mean       2.800050      4.218979      6.462676  
std        1.606239      1.125491      2.319242  
min        0.000000      0.000000      0.000000  
25%        1.877626      3.752478      4.013579  
50%        2.673698      4.611434      7.131194  
75%        3.695294      4.958306      8.314639  
max       10.654430      5.950261     11.040031

__Removing the unwanted tuples__

In [7]:
df = data.dropna(axis = 0)
df = df.reset_index(drop = True)

In [8]:
df.head()

0      AC 1      AC 2  AC 3  AC 4      AC 5      AC 6  \
0  2019-08-01 00:00:00  7.518632  8.788315   0.0   0.0  2.617045  4.079041   
1  2019-08-01 00:02:00  7.426114  8.940615   0.0   0.0  2.581625  3.781231   
2  2019-08-01 00:04:00  7.052986  9.161103   0.0   0.0  2.592095  3.800127   
3  2019-08-01 00:06:00  6.665446  9.065626   0.0   0.0  2.575639  3.772891   
4  2019-08-01 00:08:00  6.674838  9.096130   0.0   0.0  2.021472  3.155697   

       AC 7      AC 8      AC 9     AC 10     AC 11     AC 12     AC 13  \
0  2.782276  4.624447  5.222060  2.151238  1.585072  0.560373  3.142941   
1  2.529366  5.057423  5.349465  2.414715  2.168184  1.818730  3.085110   
2  2.332304  6.322521  3.995392  2.237114  3.345624  2.310409  3.132799   
3  2.596200  5.805132  3.553778  1.878356  2.737645  2.510972  3.090007   
4  2.845417  6.315060  3.042244  1.914650  2.452500  2.171791  3.097202   

      AC 14     AC 15     AC 16     AC 17     AC 18  
0  2.749470  5.417774  4.113460  3.305072  6.735981  
1  2.720484  3.302422  3.986483  3.220588  6.379500  
2  2.676861  3.539026  3.797881  3.131560  6.363475  
3  2.666604  4.607439  3.062610  3.063953  6.127366  
4  2.710572  5.332696  2.640150  2.864680  5.662474

__Splitting the "0" column into two seperate columns of date and time__

In [9]:
df["Date"] = df["0"].str.split(expand=True)[0]
df["Time"] = df["0"].str.split(expand=True)[1]

__Dropping the "0" column__

In [10]:
df.drop(["0"], inplace = True, axis = 1)

__Finding the AC that had consumed the maximum and minimum power__

In [11]:
print("Data is from : ", df.Date.min(), "to", df.Date.max())

Data is from :  2019-08-01 to 2019-09-30


In [12]:
consumption = df.sum()[:18].sort_values()

In [13]:
print("Minimum Power consuming AC:",consumption[:1])

Minimum Power consuming AC: AC 5    50800.569898
dtype: object


In [14]:
print("Maximum Power consuming AC:",consumption[-1:])

Maximum Power consuming AC: AC 18    223573.076079
dtype: object


- The dataset is from 1st August 2019 to 30th September 2019
- The AC 19 had consumned the maximum power
- The AC 5 had consumed the minimum power

In [15]:
df.head()

AC 1      AC 2  AC 3  AC 4      AC 5      AC 6      AC 7      AC 8  \
0  7.518632  8.788315   0.0   0.0  2.617045  4.079041  2.782276  4.624447   
1  7.426114  8.940615   0.0   0.0  2.581625  3.781231  2.529366  5.057423   
2  7.052986  9.161103   0.0   0.0  2.592095  3.800127  2.332304  6.322521   
3  6.665446  9.065626   0.0   0.0  2.575639  3.772891  2.596200  5.805132   
4  6.674838  9.096130   0.0   0.0  2.021472  3.155697  2.845417  6.315060   

       AC 9     AC 10     AC 11     AC 12     AC 13     AC 14     AC 15  \
0  5.222060  2.151238  1.585072  0.560373  3.142941  2.749470  5.417774   
1  5.349465  2.414715  2.168184  1.818730  3.085110  2.720484  3.302422   
2  3.995392  2.237114  3.345624  2.310409  3.132799  2.676861  3.539026   
3  3.553778  1.878356  2.737645  2.510972  3.090007  2.666604  4.607439   
4  3.042244  1.914650  2.452500  2.171791  3.097202  2.710572  5.332696   

      AC 16     AC 17     AC 18        Date      Time  
0  4.113460  3.305072  6.735981  2019-08-01  00:00:00  
1  3.986483  3.220588  6.379500  2019-08-01  00:02:00  
2  3.797881  3.131560  6.363475  2019-08-01  00:04:00  
3  3.062610  3.063953  6.127366  2019-08-01  00:06:00  
4  2.640150  2.864680  5.662474  2019-08-01  00:08:00

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34642 entries, 0 to 34641
Data columns (total 20 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   AC 1    34642 non-null  float64
 1   AC 2    34642 non-null  float64
 2   AC 3    34642 non-null  float64
 3   AC 4    34642 non-null  float64
 4   AC 5    34642 non-null  float64
 5   AC 6    34642 non-null  float64
 6   AC 7    34642 non-null  float64
 7   AC 8    34642 non-null  float64
 8   AC 9    34642 non-null  float64
 9   AC 10   34642 non-null  float64
 10  AC 11   34642 non-null  float64
 11  AC 12   34642 non-null  float64
 12  AC 13   34642 non-null  float64
 13  AC 14   34642 non-null  float64
 14  AC 15   34642 non-null  float64
 15  AC 16   34642 non-null  float64
 16  AC 17   34642 non-null  float64
 17  AC 18   34642 non-null  float64
 18  Date    34642 non-null  object 
 19  Time    34642 non-null  object 
dtypes: float64(18), object(2)
memory usage: 5.3+ MB


In [17]:
index = df.columns
print(index)

Index(['AC 1', 'AC 2', 'AC 3', 'AC 4', 'AC 5', 'AC 6', 'AC 7', 'AC 8', 'AC 9',
       'AC 10', 'AC 11', 'AC 12', 'AC 13', 'AC 14', 'AC 15', 'AC 16', 'AC 17',
       'AC 18', 'Date', 'Time'],
      dtype='object')


__Creating a new dataframe using aggregate function for finding the AC usage trends__

In [18]:
## The total consumption of each day is added to find the total consumption for the day

dk = df.groupby("Date").sum()

In [19]:
dk.head()

AC 1         AC 2         AC 3         AC 4         AC 5  \
Date                                                                          
2019-08-01  3057.995528  5059.999511  1593.947741  2325.764737  1020.175948   
2019-08-02  2808.680368  3531.951039  2044.310566  2226.627257  1157.040178   
2019-08-03  2299.044380  3030.513118  2067.223618  1340.909484   968.902005   
2019-08-04  2893.893729  3068.529241  1804.586764  1080.564477  1072.481764   
2019-08-05  2481.213702  2804.836697  1240.237944   634.250077  1167.370031   

                   AC 6         AC 7         AC 8         AC 9        AC 10  \
Date                                                                          
2019-08-01  1985.364315  1831.997810  2302.206125  1595.684806  1244.336753   
2019-08-02  1654.758755  1647.026878  1045.513994  1853.368826  1246.985911   
2019-08-03  1803.619528  2252.057113   914.561485  1948.012235  1033.231280   
2019-08-04  2490.482007  1965.427474  1098.371123  1759.855562  1189.007473   
2019-08-05  2644.557386  1733.026706  1145.953278  1809.435766   991.375489   

                  AC 11        AC 12        AC 13        AC 14        AC 15  \
Date                                                                          
2019-08-01  1268.675948  1779.175408  3304.251956  2166.439514  3028.555717   
2019-08-02  1417.147574  1963.794064  3606.126926  1652.137145  3153.683301   
2019-08-03  1545.638030  2180.344723  3397.414574  1318.043113  3156.538647   
2019-08-04  1286.969300  2123.903320  3455.182462  1549.923043  3237.312583   
2019-08-05  2347.134169  2178.167291  2906.306483  1948.156037  3487.697781   

                  AC 16        AC 17        AC 18  
Date                                               
2019-08-01  2561.444699  3268.643699  5317.220293  
2019-08-02  2424.806107  3127.669674  5474.652081  
2019-08-03  2071.551159  3091.382795  5554.673461  
2019-08-04  2025.092324  3146.422670  5777.808988  
2019-08-05  2412.693734  2654.863101  4265.814987

In [20]:
# Making line plots for day wise usage of each AC seperately 

fig = sp.make_subplots(rows = 9, cols = 2, subplot_titles=[a for a in index])
k = 0
for i in range(1,10):
    for j in range(1,3):
        fig.add_trace(go.Scatter(x = dk.index,y = dk[index[k]].values, mode = 'lines'),
                      row = i, col = j)
        k+=1
fig.update_layout(height=1500, width=1000)
fig.show()

__The AC usage trends__

__Date wise trend__

- It can be seen that from 11 August 2019 there is a continuous decrease in the usage of ACs
- Then after 24 August 2019, there is sudden increase in the use of AC's
- After that there are little ups and downs in the usage of AC's from 25th August to 30th September
- There is only one sudden drop that is on from 11th September and then there is seen a rise again from around 14th September
- AC5 and AC8 had consumed the least power, reaching a max of 2000 per day
- AC18 had consumed the most power, reaching 6000 per day
- Other ACs were in the range of 3000 to 4000 per day usage


## Conclusions

- The most used AC is AC 18
- The least used AC is AC 5
- From 1st August to 10th August the weather was very warm
- From 11th August to 23rd August the weather must be cold enough to not require the AC
- Then from 25th August there was a sudden increase of heat
- Around 12th September the was cold (might be due to rain)
- Rest the days were hot from 25th August to 30th September 2019

In [21]:
import plotly.io as pio
pio.renderers.default = "svg"